consensus/tendermint/core/core.go

```
@@ -0,0 +1,327 @@
+		seal := PrepareCommittedSeal(c.currentRoundState.GetCurrentProposalHash())
+		msg.CommittedSeal, err = c.backend.Sign(seal)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Sign message
+	data, err := msg.PayloadNoSig()
+	if err != nil {
+		return nil, err
+	}
+	msg.Signature, err = c.backend.Sign(data)
+	if err != nil {
+		return nil, err
+	}
+
+	// Convert to payload
+	payload, err := msg.Payload()
+	if err != nil {
+		return nil, err
+	}
+
+	return payload, nil
+}
+
+func (c *core) broadcast(ctx context.Context, msg *message) {
+	logger := c.logger.New("step", c.currentRoundState.Step())
+
+	payload, err := c.finalizeMessage(msg)
+	if err != nil {
+		logger.Error("Failed to finalize message", "msg", msg, "err", err)
+		return
+	}
+
+	// Broadcast payload
+	logger.Debug("broadcasting", "msg", msg.String())
+	if err = c.backend.Broadcast(ctx, c.valSet.Copy(), payload); err != nil {
+		logger.Error("Failed to broadcast message", "msg", msg, "err", err)
+		return
+	}
+}
+
+func (c *core) isProposer() bool {
+	return c.valSet.IsProposer(c.address)
+}
+
+func (c *core) commit() {
+	c.setStep(precommitDone)
+
+	proposal := c.currentRoundState.Proposal()
+
+	if proposal != nil {
+		if proposal.ProposalBlock != nil {
+			log.Warn("commit a block", "hash", proposal.ProposalBlock.Header().Hash(), "block", proposal.ProposalBlock)
+		} else {
+			log.Error("commit a NIL block",
+				"block", proposal.ProposalBlock,
+				"height", c.currentRoundState.height.String(),
+				"round", c.currentRoundState.round.String())
+		}
+
+		committedSeals := make([][]byte, c.currentRoundState.Precommits.VotesSize(proposal.ProposalBlock.Hash()))
+		for i, v := range c.currentRoundState.Precommits.Values(proposal.ProposalBlock.Hash()) {
+			committedSeals[i] = make([]byte, types.PoSExtraSeal)
+			copy(committedSeals[i][:], v.CommittedSeal[:])
+		}
+
+		if err := c.backend.Commit(*proposal.ProposalBlock, committedSeals); err != nil {
+			return
```

This error is not handled properly. Could we raise or log an error here, as in the other failure conditions?

---

done https://github.com/clearmatics/autonity/pull/185/commits/75aa7f6cbe457f222336dc1c2738742b76f6d667